src/test/java/seedu/address/logic/commands/TakeAttendanceCommandTest.java

```diff
@@ -0,0 +1,149 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.attendance.AttendancePresence;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.session.Session;
+import seedu.address.model.session.SessionNumber;
+import seedu.address.testutil.PersonBuilder;
+
+
+public class TakeAttendanceCommandTest {
+
+    @Test
+    public void execute_takeAttendanceWithSingleStudent_success() throws CommandException {
+        Model model = new ModelManager();
+        Person student = new PersonBuilder().withName("Alice").build();
+        Person newStudent = new PersonBuilder().withName("Bob").build();
+        model.addPerson(student);
+        model.addPerson(newStudent);
+        Session currentSession = new Session(new SessionNumber("2"), student);
```

Comment:
Not related to this file, but you might want to add a testcase for "floats" in SessionNumberTest.java.
I think it wasnt covered

Reply:
`public static final String VALIDATION_REGEX = "^[0-9]\\d*$";`
this line in SessionNumber:14 ensures that only numbers are allows (no decimal point), hence check for "floats" may not be necessary